In [1]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
# Get dataset
hhVicData = pd.read_csv('NCVS_HOUSEHOLD_VICTIMIZATION_1993-2019.csv')

In [3]:
hhVicData.head()

,year,weight,msa,hincome,hhage,hhgen,hhhisp,hhrace1r,hheth1r,hnumber,popsize,region,notify,newcrime,newoff,vicservices,locationr
0,1993,3219.62735,3,1,3,1,2,1,1,1,NaN,-1,2,3,8,2.0,1
1,1993,3219.62735,3,1,3,1,2,1,1,1,NaN,-1,2,3,8,2.0,1
2,1993,3219.62735,3,1,3,1,2,1,1,1,NaN,-1,2,3,8,2.0,1
3,1993,2999.05015,3,4,2,1,2,1,1,3,NaN,-1,1,3,8,2.0,1
4,1993,2999.05015,3,4,3,2,2,1,1,1,NaN,-1,2,3,8,2.0,1


In [8]:
hhVicData.shape

(216269, 17)

In [4]:
hhVicData.describe()

,year,weight,msa,hincome,hhage,hhgen,hhhisp,hhrace1r,hheth1r,hnumber,popsize,region,notify,newcrime,newoff,vicservices,locationr
count,216269.000000,216269.000000,216269.000000,216269.000000,216269.000000,216269.000000,216269.000000,216269.000000,216269.000000,216269.000000,177437.000000,216269.000000,216269.000000,216269.0,216269.000000,214994.000000,216269.000000
mean,2004.462822,2539.211250,1.731039,18.443323,3.115338,1.466673,2.208430,1.233825,1.603646,2.206262,1.598133,2.109322,1.661306,3.0,7.584027,1.983279,1.910343
std,8.361142,1954.156798,0.684604,31.214778,0.995070,0.498889,5.383467,0.532468,1.048591,0.819708,1.482455,1.714463,0.537349,0.0,0.781140,0.128226,1.272344
min,1993.000000,33.798650,1.000000,1.000000,1.000000,1.000000,1.000000,1.000000,1.000000,1.000000,0.000000,-1.000000,1.000000,3.0,6.000000,1.000000,1.000000
25%,1997.000000,1955.989810,1.000000,3.000000,2.000000,1.000000,2.000000,1.000000,1.000000,2.000000,1.000000,1.000000,1.000000,3.0,8.000000,2.000000,1.000000
50%,2003.000000,2248.094720,2.000000,5.000000,3.000000,1.000000,2.000000,1.000000,1.000000,2.000000,1.000000,3.000000,2.000000,3.0,8.000000,2.000000,1.000000
75%,2012.000000,2812.389420,2.000000,7.000000,4.000000,2.000000,2.000000,1.000000,2.000000,3.000000,2.000000,3.000000,2.000000,3.0,8.000000,2.000000,3.000000
max,2019.000000,79767.074400,3.000000,88.000000,5.000000,2.000000,88.000000,3.000000,4.000000,4.000000,5.000000,4.000000,8.000000,3.0,8.000000,2.000000,5.000000


<b> Observation </b>  
popsize and vicservices have some missing values.

In [5]:
# Find out the number of missing values for popsize and vicservices
hhVicData.isnull().sum()

year               0
weight             0
msa                0
hincome            0
hhage              0
hhgen              0
hhhisp             0
hhrace1r           0
hheth1r            0
hnumber            0
popsize        38832
region             0
notify             0
newcrime           0
newoff             0
vicservices     1275
locationr          0
dtype: int64

In [12]:
hhVicData.loc[(hhVicData['popsize'].isnull()) & (hhVicData['vicservices'].isnull())].shape

(165, 17)

<b> Observation </b>  
Out of the 216,269 noted victimisations,  
38832 have empty popsize values (~ 17%) and  
1275 have empty vicservices values (~0.6%)  
And 165 rows have both popsize and vicservices empty (~0.0008%)  